src/Codeception/Module/Db.php

```diff
@@ -288,6 +330,93 @@ protected function cleanup()
 
     protected function loadDump()
     {
+        if ($this->config['dumptool']) {
+            $this->loadDumpUsingTool();
+        } else {
+            $this->loadDumpUsingDriver();
+        }
+    }
+
+    protected function loadDumpUsingTool()
+    {
+        if (!$this->config['dump']) {
+            $this->debug("[Db] No dump file found. Skip loading the dump using the dumptool.");
+            return;
+        }
+        try {
+            $command = $this->buildDumpToolCommand($this->config['dumptool']);
+            $this->debug("[Db] Executing dumptool command: `$command`");
+
+            list($result, $output, $exitCode) = $this->execute($command);
+
+            $this->debug("[Db] Done running dumptool command with result: `$result`");
+            $this->debug("[Db] Exit code: `$exitCode`");
+            $this->debug("[Db] ".count($output)." line/s of output:\n");
+            foreach ($output as $l) {
+                $this->debug("[Db] \t$l");
+            }
+
+            if (0 !== $exitCode) {
+                throw new \RuntimeException(
+                    implode(
+                        "\n",
+                        [
+                            "The dump tool command did not end successfully: ",
+                            "Exit code: $exitCode",
+                            "Output:",
+                            implode("\n", $output),
+                        ]
+                    )
+                );
+            }
+        } catch (\Exception $e) {
+            $this->debug(implode("\n", [get_class($e), $e->getMessage(), $e->getTraceAsString()]));
+            throw new ModuleException(
+                __CLASS__,
+                implode(
+                    "\n",
+                    [
+                        $e->getMessage(),
+                        sprintf(
+                            'Attempted to load the dump `%1$s` using the tool `%2$s`',
+                            $this->config['dump'],
+                            $this->config['dumptool']
+                        ),
+                    ]
+                )
+            );
+        }
+    }
+
+    protected function buildDumpToolCommand($dumptool, $config = [], $dsn = '')
+    {
+        $config = $config ?: $this->config;
+        $dsn = $dsn ?: isset($config['dsn']) ? $config['dsn'] : '';
+        $dsnVars = [];
+        $dsnWithoutDriver = preg_replace('/^[a-z]+:/i', '', $dsn);
+        foreach (explode(';', $dsnWithoutDriver) as $item) {
+            $i = explode('=', $item);
+            if (isset($i[1])) {
+                $dsnVars[$i[0]] = $i[1];
+            }
+        }
+
+        $vars = array_merge($dsnVars, $config);
+        foreach ($vars as $key => $value) {
+            $vars['$'.$key] = $value;
+            unset($vars[$key]);
+        }
+        return str_replace(array_keys($vars), array_values($vars), $dumptool);
+    }
+
+    protected function execute($command)
```

Comment:
Well, I called it execute because the php function that is wrapping is `exec` any ideas?